Home_Work_2.py

```diff
@@ -0,0 +1,178 @@
+from pydantic import BaseModel
+from pprint import pprint
+from abc import ABC, abstractmethod
+from functools import wraps
+
+
+class BookModel(BaseModel):
+    name: str
+    author: str
+    issue_date: str
+
+
+class AbstractClass(ABC):
+    @abstractmethod
+    def __init__(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def __str__(self):
+        raise NotImplementedError
+
+
+class Book(AbstractClass):
+    name = None
+    __author = None
+    issue_date = None
+    book_list = list()
+    book_dict = dict()
+
+    def __init__(self, book: BookModel):
+        self.name = book.name
+        self.__author = book.author
+        self.issue_date = book.issue_date
+        self.book_list.append(self.__repr__())
+        self.book_dict.setdefault(self.__author, self.name)
+
+    def __str__(self):
+        return (f'Book "{self.name.title()}" was written by {self.__author.title()} '
+                f'and issued {self.issue_date}.')
+
+    def __repr__(self):
+        return f'{self.name.title()}'
+
+
+class Magazine(Book):
+
+    def __init__(self, book: BookModel, style):
```

Comment:
please add type annotation for style argument.

Reply:
Done